awscli/customizations/dynamodb/types.py

```diff
@@ -0,0 +1,290 @@
+
+        elif self._is_boolean(value):
+            dynamodb_type = BOOLEAN
+
+        elif self._is_number(value):
+            dynamodb_type = NUMBER
+
+        elif self._is_string(value):
+            dynamodb_type = STRING
+
+        elif self._is_binary(value):
+            dynamodb_type = BINARY
+
+        elif self._is_type_set(value, self._is_number):
+            dynamodb_type = NUMBER_SET
+
+        elif self._is_type_set(value, self._is_string):
+            dynamodb_type = STRING_SET
+
+        elif self._is_type_set(value, self._is_binary):
+            dynamodb_type = BINARY_SET
+
+        elif self._is_map(value):
+            dynamodb_type = MAP
+
+        elif self._is_list(value):
+            dynamodb_type = LIST
+
+        else:
+            msg = 'Unsupported type "%s" for value "%s"' % (type(value), value)
+            raise TypeError(msg)
+
+        return dynamodb_type
+
+    def _is_null(self, value):
+        if value is None:
+            return True
+        return False
+
+    def _is_boolean(self, value):
+        if isinstance(value, bool):
+            return True
+        return False
+
+    def _is_number(self, value):
+        if isinstance(value, (six.integer_types, Decimal)):
+            return True
+        elif isinstance(value, float):
+            raise TypeError(
+                'Float types are not supported. Use Decimal types instead.')
+        return False
+
+    def _is_string(self, value):
+        if isinstance(value, six.string_types):
+            return True
+        return False
+
+    def _is_binary(self, value):
+        if isinstance(value, Binary):
+            return True
+        elif isinstance(value, bytearray):
+            return True
+        elif six.PY3 and isinstance(value, six.binary_type):
+            return True
+        return False
+
+    def _is_set(self, value):
+        if isinstance(value, Set):
+            return True
+        return False
+
+    def _is_type_set(self, value, type_validator):
+        if self._is_set(value):
+            if False not in map(type_validator, value):
+                return True
+        return False
+
+    def _is_map(self, value):
+        if isinstance(value, Mapping):
+            return True
+        return False
+
+    def _is_list(self, value):
+        if isinstance(value, list):
+            return True
+        return False
+
+    def _serialize_null(self, value):
+        return True
+
+    def _serialize_bool(self, value):
+        return value
+
+    def _serialize_n(self, value):
+        number = str(DYNAMODB_CONTEXT.create_decimal(value))
+        if number in ['Infinity', 'NaN']:
+            raise TypeError('Infinity and NaN not supported')
+        return number
+
+    def _serialize_s(self, value):
+        return value
+
+    def _serialize_b(self, value):
+        if isinstance(value, Binary):
+            value = value.value
+        return value
+
+    def _serialize_ss(self, value):
```

Comment:
For sets I have a couple of questions:

1) Playing around with sets, I get this output:
```
$ aws ddb select Sample
ConsumedCapacity: null
Count: 3
Items:
- id: bar
  var: 2
- id: boo
  var: !!set
    foo: null
- id: foo
  var: 1.034
ScannedCount: 3
```
However looking at the YAML specification, it is supposed to look more like this: http://yaml.org/type/set.html with the use of `?` or curly brackets.

2) How would do we use jmespath expressions to query for elements in the set? Do you have a sample `--query` parameter you could share?

Reply:
1. That's... weird. I'll see what I can do to make the output nicer there
2. You can't index into a set, so all you can do is grab the whole thing like any other object.